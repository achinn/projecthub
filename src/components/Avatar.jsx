import React from 'react'
import PropTypes from 'prop-types'

import { User } from 'src/models'


const Avatar = ({ user, size }) => (
  <img
    alt={`@${user.login}`}
    className="avatar"
    height={size}
    width={size}
    src={user.avatar}
  />
)

Avatar.propTypes = {
  user: PropTypes.instanceOf(User).isRequired,
  size: PropTypes.number,
}

Avatar.defaultProps = {
  size: 20,
}

export default Avatar
