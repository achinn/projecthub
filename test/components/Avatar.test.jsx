import React from 'react'
import { shallow } from 'enzyme'

import Avatar from 'components/Avatar'

import { testUser } from 'test/fixtures'

describe('Avatar', () => {
  let props
  let describedComponent

  const avatar = () => {
    describedComponent = describedComponent || shallow(<Avatar {...props} />)
    return describedComponent
  }

  beforeEach(() => {
    props = { user: testUser }
    describedComponent = undefined
  })

  it('displays a single <img>', () => {
    expect(avatar().find('img')).toHaveLength(1)
    expect(avatar().html()).toMatchSnapshot()
  })

  it('has a default size', () => {
    expect(avatar().props()).toMatchObject({ height: 20, width: 20 })
  })

  it('has configurable size', () => {
    props.size = 40
    expect(avatar().props()).toMatchObject({ height: 40, width: 40 })
  })

  it('sets the alt tag to the user’s login', () => {
    expect(avatar().props()).toHaveProperty('alt', `@${props.user.login}`)
  })

  it('sets the img src to the user’s avatar', () => {
    expect(avatar().props()).toHaveProperty('src', props.user.avatar)
  })
})
