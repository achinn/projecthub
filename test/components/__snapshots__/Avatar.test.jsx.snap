// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar displays a single <img> 1`] = `"<img alt=\\"@test\\" class=\\"avatar\\" height=\\"20\\" width=\\"20\\" src=\\"test.jpg\\"/>"`;
